sickbeard/tvcache.py

```diff
@@ -171,37 +170,58 @@ def updateCache(self):
                 # set updated
                 self.setLastUpdate()
 
-                cl = []
-                for item in data['entries'] or []:
-                    ci = self._parseItem(item)
-                    if ci is not None:
-                        cl.append(ci)
+                # get last 5 rss cache results
+                recent_results = self.provider.recent_results
+                found_recent_results = 0  # A counter that keeps track of the number of items that have been found in cache
+                stop_at = 3  # Configuration as an error margin, to stop at. The lower the number, the faster it will stop parsing items
```

Comment:
Consider making `stop_at` a provider variable in case we want to make it vary by provider.